main.nf

```diff
@@ -459,6 +501,91 @@ process fastp {
     """
 }
 
+/*
+ * Remove host read contamination
+ */
+(trimmed_reads, ch_trimmed_reads_remove_host) = trimmed_reads.into(2)
+
+process host_bowtie2index {
+    tag "${genome}"
+
+    input:
+    file(genome) from ch_host_fasta
+
+    output:
+    file("bt2_index_base*") into ch_host_bowtie2index
+
+    when: params.host_fasta
+
+    script:
+    """
+    bowtie2-build --threads "${task.cpus}" "${genome}" "bt2_index_base"
+    """
+}
+
+process remove_host {
+    tag "${name}"
+
+    publishDir "${params.outdir}/QC_shortreads/remove_host/", mode: 'copy',
+        saveAs: {filename ->
+                    if (filename.indexOf("read_ids.txt") > 0) "$filename"
+                    else null
+                }
+
+    input:
+    set val(name), file(reads) from ch_trimmed_reads_remove_host
+    file(index) from ch_host_bowtie2index
+
+    output:
+    set val(name), file("${name}_host_unmapped*.fastq.gz") into ch_trimmed_reads_host_removed
+    file("${name}.bowtie2.log") into ch_host_removed_log
+    file("${name}_host_mapped*.read_ids.txt") into ch_mapped_read_ids
+
+    when: params.host_fasta || params.host_genome
+
+    script:
+    def sensitivity = params.host_removal_verysensitive ? "--very-sensitive" : "--sensitive"
+    if ( !params.single_end ) {
```

Comment:
for avoiding to duplicate most of the process code (sin gle end and paired end), you can define an input parameter, like at https://github.com/nf-core/mag/blob/master/main.nf#L672

Reply:
ok, but in this case there are multiple parts in the process code affected: 
1.  `-1 "${reads[0]}" -2 "${reads[1]}"`
2. `--un-conc-gz ${name}_host_unmapped_%.fastq.gz \`
    `--al-conc-gz ${name}_host_mapped_%.fastq.gz \`
3. `zcat ${name}_host_mapped_1.fastq.gz | awk '{if(NR%4==1) print substr(\$0, 2)}' > ${name}_host_mapped_1.read_ids.txt`
   `zcat ${name}_host_mapped_2.fastq.gz | awk '{if(NR%4==1) print substr(\$0, 2)}' > ${name}_host_mapped_2.read_ids.txt`

Maybe in this case it doesn't necessarily get cleaner if solved like this?